src/lib/support/value_parser/ValueParserComplex.h

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <charconv>
+#include <functional>
+#include <type_traits>
+#include <vector>
+
+#include <app/data-model/List.h>
+#include <app/data-model/Nullable.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/value_parser/ValueParserIntegral.h>
+
+namespace chip {
+
+template <typename T>
+CHIP_ERROR ParseValue(const char *& iter, const char * end, chip::Optional<T> & value,
```

Comment:
These parsing functions are pretty chip-tool-specific.  Why are these in `src/lib/support`, and in the toplevel `chip` namespace?

Reply:
@andy31415 asked to add unit tests for these `ParseValue` functions. I ran into tons of pigweed bugs when adding unit tests in `chip-tool` example. The main cause is failing to find `pw_unit_test_AUTOMATIC_RUNNER` but I don't have idea about how this variable is set.